### PR TITLE
[DOC] rb_ary_s_create

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Tune up Array::[].  Removes two examples that don't actually call Array::[].

### CODE REVIEW

File: array.c [hunk 1]
```diff
@@ -1159,11 +1159,11 @@ rb_ary_initialize(int argc, VALUE *argv, VALUE ary)
 }
 
 /*
- * Returns a new array populated with the given objects.
+ * Returns a new +Array+ object, populated with the given objects:
+ *
+ *   Array.[]( 1, 'a', /^A/) # => [1, "a", /^A/]
+ *   Array.[]                # => []
```
Comment (peterzhu2118) [inline array.c:1165]: Can we also keep the example using `Array[1, 'a', /^A/]`? It's the same method call but using different syntax (the most idiomatic syntax).

File: array.c [hunk 2]
```diff
@@ -1159,11 +1159,11 @@ rb_ary_initialize(int argc, VALUE *argv, VALUE ary)
 }
 
 /*
- * Returns a new array populated with the given objects.
+ * Returns a new +Array+ object, populated with the given objects:
+ *
+ *   Array.[]( 1, 'a', /^A/) # => [1, "a", /^A/]
```
Comment (peterzhu2118) [inline array.c:1164]: Nit: extra space:

```suggestion
 *   Array.[](1, 'a', /^A/) # => [1, "a", /^A/]
```

File: array.c (hunk 1 above)
Comment (BurdetteLamar) [reply array.c]: Restored (and moved the idiomatic syntax to top).

File: array.c (hunk 2 above)
Comment (BurdetteLamar) [reply array.c]: Fixed.  (There are no nits.)

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**